File: commands.py
from flask import Flask
import requests
import time

app = Flask(__name__)

# Helper function for GET
def getRequest(url):
    print(f"Performing GET request to {url}")
    response = requests.get(url)
    print(response.text)

# Helper function for POST, PUT, and DELETE
def sendDataRequest(method, url, data):
    print(f"Performing {method} request to {url}")
    response = requests.request(method, url, json=data)
    print(response.text)

@app.route('/run-transactions')
def run_transactions():
    print("Starting transactions")
    print("---------------------")
    print()

    getRequest('http://localhost:3000/transaction/get')
    time.sleep(1.5)

    print()
    print("Adding a transaction")
    print("--------------------")
    print()
    
    sendDataRequest('post', 'http://localhost:3000/transaction/post', {'id': 1, 'amount': 100, 'description': 'Test transaction'})
    time.sleep(1.5)

    getRequest('http://localhost:3000/transaction/get')
    time.sleep(1.5)

    print()
    print("Updating the transaction")
    print("------------------------")
    print()
    
    sendDataRequest('put', 'http://localhost:3000/transaction/put/1', {'amount': 200, 'description': 'Updated transaction'})
    time.sleep(1.5)

    get_request('http://localhost:3000/transaction/get')
    time.sleep(1.5)

    print()
    print("Deleting the transaction")
    print("------------------------")
    print()
    
    sendDataRequest('delete', 'http://localhost:3000/transaction/delete/1', {})
    time.sleep(1.5)

    getRequest('http://localhost:3000/transaction/get')
    time.sleep(1.5)

    return "Transactions Completed"

if __name__ == '__main__':
    app.run(debug=True)

File: commands.js
const axios = require('axios')

// Helper function for sleep
const sleep = (seconds) => new Promise(resolve => setTimeout(resolve, seconds * 1000))

// Helper function for GET
const getRequest = async (url) => {
    console.log(`\nPerforming GET request to ${url}\n`)
    try {
        const response = await axios.get(url)
        console.log(response.data)
    } catch (error) {
        console.error(`Error: ${error}`)
    }
};

// Helper function for POST, PUT, and DELETE
const sendDataRequest = async (method, url, data) => {
    console.log(`\nPerforming ${method.toUpperCase()} request to ${url}\n`)
    try {
        const response = await axios({ method, url, data, headers: {'Content-Type': 'application/json'} })
        console.log(response.data)
    } catch (error) {
        console.error(`Error: ${error}`)
    }
}

// Main function to run the transactions
const runTransactions = async () => {
    console.log("\nStarting transactions")
    console.log("---------------------")
    await getRequest('http://localhost:3000/transaction/get')
    await sleep(1.5)

    console.log("\nAdding a transaction")
    console.log("--------------------")
    await sendDataRequest('post', 'http://localhost:3000/transaction/post', { id: 1, amount: 100, description: 'Test transaction' })
    await sleep(1.5)

    await getRequest('http://localhost:3000/transaction/get')
    await sleep(1.5)

    console.log("\nUpdating the transaction")
    console.log("------------------------")
    await sendDataRequest('put', 'http://localhost:3000/transaction/put/1', { amount: 200, description: 'Updated transaction' })
    await sleep(1.5)

    await getRequest('http://localhost:3000/transaction/get')
    await sleep(1.5)

    console.log("\nDeleting the transaction")
    console.log("------------------------")
    await sendDataRequest('delete', 'http://localhost:3000/transaction/delete/1')
    await sleep(1.5)

    await getRequest('http://localhost:3000/transaction/get')
    await sleep(1.5)

    console.log("\n")
}

runTransactions()

